### PR TITLE
US126962 [Engagement] Learner Engagement Dashboard Page. Search

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US126962](https://rally1.rallydev.com/#/23525809118d/iterationstatus?detail=%2Fuserstory%2F521050018628&sharedViewId=328666453148&view=dc9e7043-1cf6-486d-a4aa-3a212976e789&fdp=true?fdp=true): [Engagement] Learner Engagement Dashboard Page

This PR includes only `search` input and its behavior. It was made to make early integration with John's branch

* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility. The search can be fired with a keyboard
  * [x] Test cases
     * The page shows skeleton list of 10 items when a user hits the search button
* Security, devops, perf: N/A
* UX and docs
  ~* [ ] Demo to Kevin~ Will demo when I finish all components of the story
  
 FYI @NicholasHallman @anhill-D2L 
  
  
image

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
Hey @johngwilkinson, there are a few `ViewState` tests failed due to `userSelection` page.
Should I fix them?

**Comment by johngwilkinson** [comment]
> Hey @johngwilkinson, there are a few `ViewState` tests that failed due to `userSelection` page.
> Should I fix them?

Sure, if you like. I might not get back to the client today. There may be other issues in my code too, I'm not sure.

**Comment by johngwilkinson** [comment]
>     * The page shows skeleton list of 10 items when a user hits the search button

To clarify: does it also, after 2s, show the hard-coded list of two users?

**Comment by Vitalii-Misechko** [comment]
> > ```
> > * The page shows skeleton list of 10 items when a user hits the search button
> > ```
> 
> To clarify: does it also, after 2s, show the hard-coded list of two users?

That's right.

File: components/user-selector.js [hunk 1]
```diff
@@ -28,6 +30,14 @@ class UserSelector extends SkeletonMixin(Localizer(MobxLitElement)) {
 				:host([hidden]) {
 					display: none;
 				}
+
+				.d2l-insights-user-selector-search {
+					display: flex;
+					flex-wrap: nowrap;
+					max-width: 334px;
```
Comment (bemailloux) [inline components/user-selector.js:37]: What does `334px` correspond to?

**Review by johngwilkinson — APPROVED** [review]
Looks good. I'm starting to test the API now, FYI.

**Review by bemailloux — APPROVED** [review]

File: components/user-selector.js (hunk 1 above)
Comment (Vitalii-Misechko) [reply components/user-selector.js:37]: I've copied it from `d2l-insights-tree-selector-search`

**Pull request merged**